src/transformers/generation/logits_process.py

```diff
@@ -1148,3 +1148,114 @@ def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor) -> to
         cond_logits, uncond_logits = scores.split(unguided_bsz, dim=0)
         scores = uncond_logits + (cond_logits - uncond_logits) * self.guidance_scale
         return scores
+
+
+class UnbatchedClassifierFreeGuidanceLogitsProcessor(LogitsProcessor):
+    r"""Logits processor for Classifier-Free Guidance (CFG). The processors
+    computes a weighted average across scores from prompt conditional and prompt unconditional (or negative) logits,
+    parameterized by the `guidance_scale`. The unconditional scores are computed internally by prompting `model` with
+    the `unconditional_ids` branch.
+
+    See [the paper](https://arxiv.org/abs/2306.17806) for more information.
+
+    Args:
+        guidance_scale (float):
```

Comment:
(nit) could you make sure that the types for all the args are decorated as code? The tensor types are fine here, but the floats and ints need decorating for all of the args
```suggestion
        guidance_scale (`float`):
```